src/librustc_resolve/lib.rs

```diff
@@ -2319,20 +2319,26 @@ impl<'a> Resolver<'a> {
                     } else if i == 0 {
                         (format!("use of undeclared type or module `{}`", ident), None)
                     } else {
-                        let mut msg =
-                            format!("could not find `{}` in `{}`", ident, path[i - 1].ident);
+                        // Do not emit `PathRoot`(`{{root}}`) as it's unclear.
+                        let ident_name = &format!("`{}`", path[i - 1].ident);
+                        let ident_name = if path[i - 1].ident.name == kw::PathRoot {
+                            "the crate root"
+                        } else {
+                            ident_name
+                        };
```

Comment:
```suggestion
                        // Do not emit `PathRoot`(`{{root}}`) as it's unclear.
                        let ident_str;
                        let ident_name = if path[i - 1].ident.name == kw::PathRoot {
                            "the crate root"
                        } else {
                            ident_str = format!("`{}`", path[i - 1].ident);
                            &ident_str
                        };
```
Something like this is cleaner imo

Reply:
or Cow.